### ISSUE TITLE
GDScript 2.0: Cannot document inner class with doc comments

### ISSUE THREAD

**Issue by Yagich** [godotengine/godot#70289]
### Godot version

4.0b8

### System information

Linux/Manjaro

### Issue description

The [documentation page](https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_documentation_comments.html) for doc comments mentions that you can document inner classes:
```gdscript
## Documenting an inner class.
##
## The same rules apply apply here. The documentation must
## immediately precede the class definition.
##
## @tutorial: http://the/tutorial/url.com
class Inner:

    ## Inner class variable v4.
    var v4

    ## Inner class function fn.
    func fn(): pass
```
However, after having done that, no documentation page is created for the inner class.

### Steps to reproduce

1. Create a `class_name`d script with an inner class;
2. Try to document the inner class and its properties.
3. Observe that you can find the documentation page for the outer class, but the inner class can not be found anywhere.

### Minimal reproduction project

[inner-doc-comments.zip](https://github.com/godotengine/godot/files/10256732/inner-doc-comments.zip)

**Comment by dustinsth** [comment]
I confirmed that this is a regression and affects win64 builds.
The regression began in 4.0beta5 and is still persistent in the latest build.

**Comment by dustinsth** [comment]
It seems this is the commit where the regression first appears: https://github.com/godotengine/godot/commit/bce6f1792ebce5de31700e72f57830e5d87b419a

**Comment by akien-mga** [comment]
CC @rune-scape

**Comment by dustinsth** [comment]
In the commit [this portion of code](https://github.com/godotengine/godot/pull/68374/files#diff-2d47041c61b0e94d891650482b0cf9398f22729abdc811f94c2b7da902ed392fL632-L635) was removed from the end of `update_doc()` in `gdscript.cpp`:
```
for (KeyValue<StringName, Ref<GDScript>> &E : subclasses) {
	E.value->_update_doc();
}
```

When I restored this code the Documentation for the inner classes worked again.

I'm still learning how the GDScript module works, so I'm not sure yet if restoring this has any feature/performance regressions.
@rune-scape might be able to provide more context on the change.
